### ISSUE TITLE
Save secret key after entering it into wallet but before entering a password

### ISSUE THREAD

**Issue by falkonprods** [blockstack/blockstack-app#2191]
When re-instantiating the wallet with an existing secret key, the flow used to go like this:

1. Enter the secret key into wallet window
2. Return to browser
3. Re-open wallet, prompted to enter a password or start over with the secret key

This flow gave the opportunity to create a secure password in a password manager in the browser, then return to the wallet and enter it.

However, now the flow doesn't allow this, instead it does this:

1. Enter secret key into wallet window
2. Return to browser
3. Re-open wallet, have to start over with the secret key

Would be ideal to have the original flow.

**Comment by sjwhitely** [comment]
+1 on this - got good feedback from a user on Discord about this same thing. https://discord.com/channels/621759717756370964/621759718192447502/940867396925788200

**Comment by fbwoolf** [comment]
> +1 on this - got good feedback from a user on Discord about this same thing. https://discord.com/channels/621759717756370964/621759718192447502/940867396925788200

Easy work around here is just to onboard via the full page view until this is addressed in the extension popup. This should be resolved soon in the popup.

**Comment by andresgalante** [comment]
@fbwoolf if this will be resolved in `3.3` can we move it to "ready to ship" and link the PR that's fixing it please?

**Comment by fbwoolf** [comment]
> @fbwoolf if this will be resolved in `3.3` can we move it to "ready to ship" and link the PR that's fixing it please?

I am not 100% sure yet if Kyran's work fixed this. @kyranjamie do you know if your changes to routing with the key mgmt refactor might have fixed this? I know you simplified things with the initial route after refactoring the vault reducer.

**Comment by kyranjamie** [comment]
This is not addressed by my PR. With these changes, the key is saved in the frame's memory _until_ the point at which it is encrypted with a user's password, and thus it is lost when the frame closes.

To achieve this, we'll need to reuse the mechanism of persisting in memory keys for the pre-wallet creation step as well. This might be a good one to address before releasing my PR, though the workaround is quite simple, just using a full mode (we could enforce this 🤔 )

**Comment by fbwoolf** [comment]
This does work in full mode as a workaround. I was thinking some of the conditionals I had to use to keep the initial popup route in sync were causing the popup to land back on the welcome page. I know you removed those and did some refactoring in the account gate, so was hoping those changes fixed this.

**Comment by fbwoolf** [comment]
@kyranjamie I rebased my PR on your branch and I do think this is fixed. The account gate now will land the user back on the page to save their secret key (but that is ok it is the same) and they can easily just move to set the password. The bug here is that it was returning the user to the welcome page and regenerating a new secret key. I think we can link this issue to your PR so it closes when your work is merged?

**Comment by kyranjamie** [comment]
> The bug here is that it was returning the user to the welcome page and regenerating a new secret key. 

Sure, but it still needs to be fixed in my PR right?

**Comment by fbwoolf** [comment]
> 
> Sure, but it still needs to be fixed in my PR right?

You are right, I was thinking this was on creating a new wallet but it is signing in with an existing secret key. So, it lands a user on `Back up your Secret Key` which needs to be fixed on your branch so the user stays on `Set Password`. Should I assign this to you and link to your PR?

**Comment by fbwoolf** [comment]
@kyranjamie I can branch off your work and try to fix this if helps?

EDIT: Started Ledger modals.

**Comment by andresgalante** [comment]
This issue should be fixed once we merge #2266
